dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -683,12 +693,13 @@ private boolean createWorkflowsAndVersionsFromDockstoreYml(List<? extends Workfl
                         final List<YamlAuthor> yamlAuthors = wf.getAuthors();
 
                         Workflow workflow = createOrGetWorkflow(workflowType, repository, user, workflowName, wf, gitHubSourceCodeRepo);
+                        workflow.getUsers().addAll(otherUsers.stream().map(User::getId).map(userDAO::findById).toList()); // Because of Hibernate transactions, safest to lookup users again
                         WorkflowVersion version = addDockstoreYmlVersionToWorkflow(repository, gitReference, dockstoreYml, gitHubSourceCodeRepo, workflow, latestTagAsDefault, yamlAuthors);
 
                         // Create some events.
                         eventDAO.createAddTagToEntryEvent(user, workflow, version);
```

Comment:
If the sender doesn't have a Dockstore account, do we want to attribute/connect certain things to one of the other users that _does_ have a Dockstore account?  As written, this code will create "userless" events if the sender doesn't have a Dockstore account (I think).

Reply:
I could see the argument both ways... I feel this is a little different because the sender is associated with the event proper, whereas what I'm doing elsewhere is looking for other owners of the workflow within the event.

Also, there is no guarantee we'll find a user. And, if there is more than one possible user, which one do we pick? It seems like there is no good answer there.

Reply:
> I could see the argument both ways... I feel this is a little different because the sender is associated with the event proper, whereas what I'm doing elsewhere is looking for other owners of the workflow within the event.
> 
> Also, there is no guarantee we'll find a user. And, if there is more than one possible user, which one do we pick? It seems like there is no good answer there.

Imagine the case where the sender is a non-Dockstore-user bot, and the author is a Dockstore user.  Maybe we'd want the initiator of the above `ADD_VERSION_TO_ENTRY` Event to be the user, rather than null?

Re more than one possible user, seems like you could come up with some rules as to who to pick.  Maybe in descending precedence, author, committer, sender?  And if there's multiple of one of those categories (pretty rare, I'm guessing), pick one arbitrarily? 

Related: previously, I don't think we've create userless Events of certain types, we should probably check the UI to make sure it can display them properly.

Reply:
One other option, we could also add a column to the lambda event for committer, sender, author, etc. Then create a record for each of the three types of author. 
Current UI could get the sender as it does now. 
Future UI could take advantage of the extra info for debugging.

Reply:
There will be exact 1 sender, as well as  N authors and committers.  Every commit will have both an author and committer, although they might be the identical user.

Per https://docs.github.com/en/webhooks/webhook-events-and-payloads#push, the commits array can have up to 2048 commits, and I've seen 0 commits in there as well.

Do we want to record all of that? And if yes, for this release or as future work?

Reply:
Yikes, I didn't think through the N aspect. 
I think capture this for future work since the sender is working as it used to already

Reply:
Confirming:
When there's a bot (non-Dockstore user) pusher, the current PR behavior is to create a regular `Event` with type `EventType.ADD_VERSION_TO_ENTRY` and no/null initiator user.   That's the desired behavior?

Reply:
> Confirming: When there's a bot (non-Dockstore user) pusher, the current PR behavior is to create a regular `Event` with type `EventType.ADD_VERSION_TO_ENTRY` and no/null initiator user. That's the desired behavior?

Yes, in my opinion, given the current design. It doesn't seem right to me that the initiator could sometimes be the sender, sometimes a committer. I saw your proposal for a ranking system above (sorry, I didn't reply to that one directly), but it just seems too inconsistent to me. It would no longer consistently represent the initiator of the event, but somebody, potentially vaguely associated with the push, e.g., I fixed a type and nothing else 4 months ago.

Reply:
For the sake of noting scenarios that should be checked in the UI, this PR may cause a regular `Event` of type `PUBLISH_ENTRY` to be created with no initiator user.

Reply:
Ah good point. I went and checked the UI code, and all the initiatorId usages are guarded with `ngIf` or `?`. So I think we should be good there. 

Also, interestingly, for the unpublish events in a recent DB snapshot, about 10% have no initiator user.

But ultimately, I wouldn't be surprised if some (hopefully minor) issue arises with this, unfortunately. But it should be better than not having the workflow appear at all, at least that's the hope.